src/quo/components/settings/category/settings/view.cljs

```diff
@@ -15,10 +15,12 @@
        :size   :paragraph-2
        :style  (style/label props)}
       label])
-   [rn/flat-list
-    {:data      data
-     :style     (style/settings-items props)
-     :render-fn settings-item/view
-     :separator [rn/view {:style (style/settings-separator props)}]}]])
```

Comment:
I've always wondered if our flat-list wrapper is properly transforming the `render-fn` in terms of performance.

Also, that wrapper is doing many things. Have we considered if that is the part of the problem? We use this wrapper a lot, and it's involved in many performance issues because of flat-list, so it's worth checking (or did we already check?).

Reply:
One thing we certainly don't do is avoid regenerating anonymous functions. For our lower-level components we can do better using something like the factory functions explained in https://github.com/day8/re-frame/blob/39adca93673f334dc751ee2d99d340b51a9cc6db/docs/on-stable-dom-handlers.md#more-advanced-again

Reply:
Thanks for replying to this comment also in the other PR!  :)

I read that in the past, about a fabric of fabrics, I found it a little confusing and probably hard to maintain, but it works.

It'd be good to revisit our flat list implementation then :) - and if we apply that performance technique, make sure it's well documented -

Reply:
I agree with you, the factory function technique should be used only in critical places. To me the infrastructure/abstractions we build should be as optimal as possible because they're reused over and over, and the `flat-list` component is one such place.

Thanks for raising your concern, it's something I don't remember seeing being discussed in the last year or so.